### ISSUE TITLE
Pearl broken in latest version on macOS Mojave

### ISSUE THREAD

**Issue by yonkeltron** [pearl-core/pearl#93]
Hi,

I just upgraded `pearl` and it seems that it broke entirely. Once I got the upgrade to succeed and managed to run a `pearl init` again, I now get an error whenever invoking `pearl`:

```
Traceback (most recent call last):
  File "/Users/USERNAME/.local/share/pearl/bin/pearl", line 5, in <module>
    from pearllib.pearl import pearl
ImportError: No module named pearllib.pearl
```

Some info about my environment:

```bash
$ python3 --version
Python 3.7.6
$ bash --version
GNU bash, version 5.0.11(1)-release (x86_64-apple-darwin18.6.0)
$ uname -a
Darwin HOSTNAME 18.7.0 Darwin Kernel Version 18.7.0: Sun Dec  1 18:59:03 PST 2019; root:xnu-4903.278.19~1/RELEASE_X86_64 x86_64
```

Please let me know if I can provide any additional detail.

Thanks,
+Jonathan

**Comment by fsquillace** [comment]
Hi,

Apologize for that. I think it has to do with the major Pearl change to version 2.0.0. More details in the [Changelog](https://github.com/pearl-core/pearl/blob/master/CHANGELOG.md#201---2020-01-11).

Take a look at the [migration page](https://github.com/pearl-core/pearl/wiki/Migration-to-Pearl-version-2) to see how you can move to the new Pearl.

Good luck!

Filippo

**Comment by yonkeltron** [comment]
Hey thanks for the tip. I'm not getting a new error:

```
Traceback (most recent call last):
  File "Library/Python/3.7/bin/pearl", line 13, in <module>
    main()
  File "Library/Python/3.7/bin/pearl", line 9, in main
    pearl(sys.argv[1:])
  File "/Users/USERNAME/Library/Python/3.7/lib/python/site-packages/pearllib/pearl.py", line 68, in pearl
    env_initialized=False if args.command == 'init' else True
  File "/Users/USERNAME/Library/Python/3.7/lib/python/site-packages/pearllib/pearlenv.py", line 116, in __init__
    self._packages = self._load_packages(update_repos, verbose)
  File "/Users/USERNAME/Library/Python/3.7/lib/python/site-packages/pearllib/pearlenv.py", line 175, in _load_packages
    pearl_conf = self.load_conf(config_filename)
  File "/Users/USERNAME/Library/Python/3.7/lib/python/site-packages/pearllib/pearlenv.py", line 192, in load_conf
    exec(config_filename.open().read(), {}, local_dict)
  File "<string>", line 5, in <module>
NameError: name 'PEARL_REPOS' is not defined
```

Anything I should be doing?

**Comment by fsquillace** [comment]
Oh right, I think is reading the old `pearl.conf` file. Try to do:

```
# backup the previous file:
mv ~/.config/pearl/pearl.conf ~

# generate the new pearl.conf file:
pearl init
```

**Comment by fsquillace** [comment]
You can even remove the entire directory `~/.config/pearl/` if does not contain relevant files anymore.

**Comment by yonkeltron** [comment]
Thanks for the candor. These changes broke a lot of things for a lot of people. We might need to abandon Pearl in favor of other solutions.

**Comment by fsquillace** [comment]
You are right. This change has been quite big. Some of them were inevitable in order to simplify the maintenance of the codebase.
The migration process could have been much smoother than the way it was. The problem is that I have limited time to work on this and smoother migration takes much effort to me.

One thing to consider is that once the migration is done properly, there should not be huge difference with previous Pearl versions.
For instance, apart from changing the file name from `install.sh` to `hooks.sh`, the package structure remained the same. So, if you own a package you are minimally affected by this.

**Comment by yonkeltron** [comment]
Hey, I want to be really clear that I wasn't complaining or asking for free support. I really respect this project and really respect your efforts to develop it. Thanks for your hard work!

Unfortunately, due to upgrade issues, we had to remove Pearl from our toolchain and switch to alternative means of config management. We eventually chose Ansible as it integrates nicely with Homebrew in a few different ways. It's not ideal (it's no Pearl) but it'll have to do for now.

We really enjoyed our time with Pearl and look forward to seeing it continue to grow. Perhaps we can revisit things in the future.

Is it alright with you if I close this issue and get it off your plate?

Thanks again for all the efforts, time, blood, sweat, and tears.
